txpipe/masks.py

```diff
@@ -61,7 +69,59 @@ def run(self):
         pix = np.where(mask)[0]
         mask = mask[pix].astype(float)
 
-        with self.open_output("mask", wrapper=True) as f:
+        return pix, mask, metadata
+
 
+
+
+class TXSimpleMaskFrac(TXSimpleMask):
+    """
+    Make a simple mask using a depth cut and bright object cut
+    Include the fractional coverage of each pixel using a high-res survey property map (e.g. N-exposures for given band(s) )
+    
+    #NOTE: This assumes all cuts to the mask are being done within TXpipe at the config Nside
+    #if we want to apply cuts on the survey property map nside (higher resolution), we will need to add this feature
+    """
+    name = "TXSimpleMaskFrac"
+    parallel = False
+    # make a mask from the auxiliary maps
+    inputs = [("aux_lens_maps", MapsFile),
+    ]
+    outputs = [("mask", MapsFile)]
+    config_options = {
+        "depth_cut": 23.5,
+        "bright_object_max": 10.0,
+        "supreme_map_file": None,
```

Comment:
I can't recall but I think it might be better to set the default to an empty string instead of None so that argparse knows what kind of type to expect.

Reply:
Wait, think I confused myself on this - is this a required parameter or an optional one?  If it's mandatory you can set the default to `str`

Reply:
yes its required. I've now set it to str